install/infra/modules/eks/database.tf

```diff
@@ -0,0 +1,54 @@
+resource "random_password" "password" {
+  count = var.enable_external_database ? 1 : 0
+
+  length           = 16
+  special          = true
+  override_special = "!#$%&*()-_=+[]{}<>:?"
+}
+
+resource "aws_db_subnet_group" "gitpod_subnets" {
+  count = var.enable_external_database ? 1 : 0
+
+  name       = "db-sg-${var.cluster_name}"
+  subnet_ids = [module.vpc.public_subnets[2], module.vpc.public_subnets[3]]
+}
+
+resource "aws_security_group" "rdssg" {
+  count = var.enable_external_database ? 1 : 0
+
+  name   = "dh-sg-${var.cluster_name}"
+  vpc_id = module.vpc.vpc_id
+
+  ingress {
+    from_port   = 0
+    to_port     = 3306
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+}
+
+resource "aws_db_instance" "gitpod" {
+  count = var.enable_external_database ? 1 : 0
+
+  allocated_storage      = 10
+  max_allocated_storage  = 100
+  engine                 = "mysql"
+  engine_version         = "5.7"
+  instance_class         = "db.t3.micro"
```

Comment:
I don't think the database setup here matches the reference architecture. This is what we have in the reference architecture: 
```
aws rds create-db-instance \
    --db-name gitpod \
    --db-instance-identifier gitpod-instance \
    --db-instance-class db.m5.large \
    --db-subnet-group-name gitpod-rds \
    --vpc-security-group-ids sg-0e538ccac25bb1387 \
    --multi-az \
    --engine mysql \
    --allocated-storage 20 \
    --max-allocated-storage 120 \
    --master-username gitpod \
    --master-user-password $MYSQL_GITPOD_PW \
    --engine-version 5.7 \
    --tags Key=project,Value=gitpod Key=Name,Value=Gitpod-MySQLDB,Key=department,Value=demo
```
([sourc](https://www.gitpod.io/docs/self-hosted/latest/reference-architecture/single-cluster-ref-arch#database)e)